ambry-server/src/main/java/com/github/ambry/server/AmbryServerRequests.java

```diff
@@ -434,10 +434,14 @@ private ServerErrorCode handleStartStoreRequest(PartitionId partitionId) {
       logger.error("Could not enable replication on {}", partitionIds);
       return ServerErrorCode.Unknown_Error;
     }
-    // startBlobStore should guarantee that store is started, so return value shouldn't be null
-    Store store = storeManager.getStore(partitionId);
-    // 4. reset partition state if it's offline (will trigger state transition)
-    if (store.getCurrentState() == ReplicaState.OFFLINE && clusterParticipant != null) {
+    // 4. reset partition state if it's in error state (will trigger state transition)
+    boolean isLocalStoreInErrorState =
+        partitionId.getReplicaIdsByState(ReplicaState.ERROR, currentNode.getDatacenterName())
+            .stream()
+            .map(r -> r.getDataNodeId().getHostname())
```

Comment:
nit: can change this code slightly to not require the intermediate set:
```java
boolean isLocalStoreInErrorState =
    partitionId.getReplicaIdsByState(ReplicaState.ERROR, currentNode.getDatacenterName())
        .stream()
        .anyMatch(r -> r.getDataNodeId().getHostname().equals(currentNode.getHostname()));
```

Reply:
Yeah, this seems better. Let me make the change.